cylc/flow/etc/job.sh

```diff
@@ -109,6 +109,7 @@ cylc__job__main() {
     export CYLC_SUITE_FINAL_CYCLE_TIME="${CYLC_SUITE_FINAL_CYCLE_POINT}"
     export CYLC_TASK_CYCLE_TIME="${CYLC_TASK_CYCLE_POINT}"
     export CYLC_TASK_WORK_PATH="${CYLC_TASK_WORK_DIR}"
+    # TODO: this in subshell? (it moved up here after Tomek's made his branch?)
     # Env-Script
     cylc__job__run_inst_func 'env_script'
```

Comment:
This hasn't moved after I've made #3440, but rather I moved it in that PR to keep it separated in a subshell with all the rest of user code.  The only code between here and user scripts is cylc message started, export some env vars, make some dirs and `cd "${CYLC_TASK_WORK_DIR}"` - nothing that couldn't execute before `env_script` I'd think.

Reply:
Yes, agreed, I can't see any sensible reason for env_script to be up there.